src/lib/falcor/falcorUtils.js

```diff
@@ -123,11 +122,69 @@ export function expandCache(cache) {
     }, cache);
   }
 
+  function assignByPath(path, value) {
+    // If using dot notation obj.key.key.key
+    if (typeof path === "string") {
+      path = path.split('.');
+    }
+    // Parent also works for array length 1, aka initial keys
+    // Parent and Key variables are used for assigning new values later
+    const parent = followPath(path.slice(0, path.length-1));
+    const key = path[path.length-1];
+    // The following key exists as it was pushed on to stack as a valid key
+    parent[key] = value;
+  }
+
   function isObject(val) {
+    // We don't count arrays as objects here. This is to protect ourselves against an expanded atom
+    // This does still leave us vulnerable to an expanded object though, but in by far most cases
+    // it would be very bad form to put an object in an atom, so this is not supported at this time.
     if (val === null || (val instanceof Array)) return false;
     return typeof val === "object";
   }
 
+  function handleRef(pathArray, refPath) {
+    const S = new Set();
+    S.add(pathArray.join('.'));
```

Comment:
a path represented as an array, that would be a string path if you did a Array.prototype.join('.') on it. I just wanted to make it clear it can't be any kind of path.

I could of course also just put in some if else statements to check if it's a string already. That would make the function more versatile.